ledger/src/ui/test/signer_authorization/test_signer_authorization.c

```diff
@@ -0,0 +1,449 @@
+    // OP_SIGAUT_SIGN + G_valid_signature[0]
+    SET_APDU("\x80\x81\x02\xd1\x98\x62\x0e\x60\x25\x5a\xd0\xb3\xc4\x1f\x0b\x0c"
+             "\x7f\x7e\xd4\x94\xa9\xfe\x45\x29\xe8\x9b\xe0\x77\xb3\x95\x87\x53"
+             "\xd7\xad\x93\xed\xa2\x7e\x02\x8a\x9f\x3e\x7d\x9a\x07\xd5\x29\xe3"
+             "\x8b\xf5\xb2\xb1\x26\x86\x0a\x02\xed\x8f\xda\x83\xb6\xe2\x1d\xd3"
+             "\xac\x65\x56\xe0\x53\x8f\x2e\xa8\x05\x4a\x8c\x43\x26\x8a",
+             rx);
+    assert(4 == do_authorize_signer(rx, &G_sigaut_ctx));
+    assert(RES_SIGAUT_MORE == APDU_DATA_PTR[0]);
+
+    // Send invalid signature with valid pubkey
+    G_sigaut_ctx.pubkey.W_len = PUBKEYSIZE;
+    memcpy(G_sigaut_ctx.pubkey.W, G_authorizers_pubkeys[0], PUBKEYSIZE);
+    // OP_SIGAUT_SIGN + invalid signature
+    SET_APDU("\x80\x81\x02\x01\x02\x03\x04\x60\x25\x5a\xd0\xb3\xc4\x1f\x0b\x0c"
+             "\x7f\x7e\xd4\x94\xa9\xfe\x45\x29\xe8\x9b\xe0\x77\xb3\x95\x87\x53"
+             "\xd7\xad\x93\xed\xa2\x7e\x02\x8a\x9f\x3e\x7d\x9a\x07\xd5\x29\xe3"
+             "\x8b\xf5\xb2\xb1\x26\x86\x0a\x02\xed\x8f\xda\x83\xb6\xe2\x1d\xd3"
+             "\xac\x65\x56\xe0\x53\x8f\x2e\xa8\x05\x4a\x8c\x43\x26\x8a",
+             rx);
+    assert(4 == do_authorize_signer(rx, &G_sigaut_ctx));
+    assert(RES_SIGAUT_MORE == APDU_DATA_PTR[0]);
+}
+
+void test_op_sigaut_get_auth_count() {
+    printf("Test OP_SIGAUT_GET_AUTH_COUNT...\n");
+
+    unsigned int rx;
+    reset_signer_authorization(&G_sigaut_ctx);
+    // OP_SIGAUT_GET_AUTH_COUNT
+    SET_APDU("\x80\x81\x03", rx);
+    assert(4 == do_authorize_signer(rx, &G_sigaut_ctx));
+    assert(3 == APDU_DATA_PTR[0]);
+}
+
+void test_op_sigaut_get_auth_at() {
+    printf("Test OP_SIGAUT_GET_AUTH_AT...\n");
+
+    unsigned int rx;
+    reset_signer_authorization(&G_sigaut_ctx);
+
+    // OP_SIGAUT_GET_AUTH_AT[0]
+    SET_APDU("\x80\x81\x04\x00", rx);
+    assert(TX_FOR_DATA_SIZE(sizeof(G_authorizers_pubkeys[0])) ==
+           do_authorize_signer(rx, &G_sigaut_ctx));
+    ASSERT_MEMCMP(APDU_DATA_PTR,
+                  G_authorizers_pubkeys[0],
+                  sizeof(G_authorizers_pubkeys[0]));
+
+    // OP_SIGAUT_GET_AUTH_AT[1]
+    SET_APDU("\x80\x81\x04\x01", rx);
+    assert(TX_FOR_DATA_SIZE(sizeof(G_authorizers_pubkeys[1])) ==
+           do_authorize_signer(rx, &G_sigaut_ctx));
+    ASSERT_MEMCMP(APDU_DATA_PTR,
+                  G_authorizers_pubkeys[1],
+                  sizeof(G_authorizers_pubkeys[1]));
+
+    // OP_SIGAUT_GET_AUTH_AT[2]
+    SET_APDU("\x80\x81\x04\x02", rx);
+    assert(TX_FOR_DATA_SIZE(sizeof(G_authorizers_pubkeys[2])) ==
+           do_authorize_signer(rx, &G_sigaut_ctx));
+    ASSERT_MEMCMP(APDU_DATA_PTR,
+                  G_authorizers_pubkeys[2],
+                  sizeof(G_authorizers_pubkeys[2]));
+}
+
+// Note: this test trusts in values written to NVM during signer authorization,
+//       so it must be run after test_op_sigaut_sign
+void test_is_authorized_signer() {
```

Comment:
Like I mentioned before, maybe we can remove the `static` keyword from the NVM variable to be able to manipulate it here, given that there's no unnecessary risks taken.